contracts/service-registry/src/client.rs

```diff
@@ -0,0 +1,169 @@
+            .change_context_lazy(|| Error::QueryActiveVerifiers {
+                service_name,
+                chain_name,
+            })
+    }
+
+    pub fn service(&self, service_name: String) -> Result<Service> {
+        self.client
+            .query(&QueryMsg::Service {
+                service_name: service_name.clone(),
+            })
+            .change_context_lazy(|| Error::QueryService(service_name))
+    }
+
+    pub fn verifier(&self, service_name: String, verifier: String) -> Result<Verifier> {
+        self.client
+            .query(&QueryMsg::Verifier {
+                service_name: service_name.clone(),
+                verifier: verifier.clone(),
+            })
+            .change_context_lazy(|| Error::QueryVerifier {
+                service_name,
+                verifier,
+            })
+    }
+}
+
+#[cfg(test)]
+mod test {
+
+    use cosmwasm_std::testing::{mock_dependencies, mock_env, mock_info, MockQuerier};
+    use cosmwasm_std::{from_json, Addr, DepsMut, QuerierWrapper, WasmQuery};
+    use router_api::ChainName;
+
+    use crate::client::{Client, Error};
+    use crate::contract::{instantiate, query};
+    use crate::msg::{InstantiateMsg, QueryMsg};
+
+    #[test]
+    fn query_active_verifiers_error() {
+        let (querier, _, addr) = setup();
+        let client: Client = client::Client::new(QuerierWrapper::new(&querier), addr).into();
+        let service_name = "verifiers".to_string();
+        let chain_name: ChainName = "ethereum".try_into().unwrap();
+        let res = client.active_verifiers(service_name.clone(), chain_name.clone());
+        assert!(res.is_err());
+
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            Error::QueryActiveVerifiers {
+                service_name,
+                chain_name
+            }
+            .to_string(),
+        );
+    }
+
+    #[test]
+    fn query_verifier_error() {
+        let (querier, _, addr) = setup();
+        let client: Client = client::Client::new(QuerierWrapper::new(&querier), addr).into();
+        let service_name = "verifiers".to_string();
+        let verifier = Addr::unchecked("verifier").to_string();
+        let res = client.verifier(service_name.clone(), verifier.clone());
+        assert!(res.is_err());
+
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            Error::QueryVerifier {
+                service_name,
+                verifier: verifier.to_string()
+            }
+            .to_string()
+        );
+    }
+
+    #[test]
+    fn query_service_error() {
+        let (querier, _, addr) = setup();
+        let client: Client = client::Client::new(QuerierWrapper::new(&querier), addr).into();
+        let service_name = "verifiers".to_string();
+        let res = client.service(service_name.clone());
+        assert!(res.is_err());
+
+        assert_eq!(
+            res.unwrap_err().to_string(),
+            Error::QueryService(service_name).to_string()
+        );
+    }
```

Comment:
consider comparing with a golden files. That might make adding the non-error cases easier too